### PR TITLE
Try/catch added to con reading to prevent crashing

### PR DESCRIPTION
**Pull request by TheFatBastid**
Without this, when an "access denied" folder is scanned, it will throw an IO error and end the scanning, and thus no cache is created.

### CODE REVIEW

File: Assets/Script/UI/MusicLibrary/SongSelection.cs [hunk 1]
```diff
@@ -60,7 +60,7 @@ public class SongSelection : MonoBehaviour {
 					_selectedIndex -= _songs.Count;
 				}
 
-				if (_songs[value] is SongViewType song) {
+				if (_songs[_selectedIndex] is SongViewType song) {
```
Comment (EliteAsian123) [inline Assets/Script/UI/MusicLibrary/SongSelection.cs:63]: Is this an intentional change?

File: Assets/Script/Song/Scanning/SongScanThread.cs [hunk 2]
```diff
@@ -190,45 +190,48 @@ public class SongScanThread {
...
 					}
 				}
-			}
+				string[] subdirectories = Directory.GetDirectories(subDir);
 
-			string[] subdirectories = Directory.GetDirectories(subDir);
-
-			foreach (string subdirectory in subdirectories) {
-				if(subdirectory != Path.Combine(subDir, "songs_updates"))
+        foreach (string subdirectory in subdirectories) {
```
Comment (EliteAsian123) [inline Assets/Script/Song/Scanning/SongScanThread.cs:229]: This line uses spaces as indentation

**Review by EliteAsian123 — CHANGES_REQUESTED** [review]

File: Assets/Script/UI/MusicLibrary/SongSelection.cs (hunk 1 above)
Comment (TheFatBastid) [reply Assets/Script/UI/MusicLibrary/SongSelection.cs:63]: Yes, this fixes an array out of bounds error when scrolling past the upper and lower limits of the song select screen. "Value" is not changed whereas _selectedIndex is clamped correctly when scrolling.

File: Assets/Script/Song/Scanning/SongScanThread.cs (hunk 2 above)
Comment (TheFatBastid) [reply Assets/Script/Song/Scanning/SongScanThread.cs]: Whoops, okay changed

**Review by EliteAsian123 — APPROVED** [review]

**Comment by EliteAsian123** [comment]
Thanks!

**Pull request merged**